src/python/platforms/fuchsia/device.py

```diff
@@ -16,6 +16,142 @@
+  beginning Fuchsia fuzzing tasks.
+  This initialization routine assumes that the GCS bucket contains the
+  standard Fuchsia SDK, as well as:
+  * /qemu-for-fuchsia/*
+  * /.ssh/* """
+  # First download the Fuchsia resources locally.
+  fuchsia_resources_dir = initialize_resources_dir()
+
+  # Then, save paths for necessary commands later.
+  qemu_path = os.path.join(fuchsia_resources_dir, 'qemu-for-fuchsia', 'bin',
+                           'qemu-system-x86_64')
+  os.chmod(qemu_path, 0o550)
+  kernel_path = os.path.join(fuchsia_resources_dir, 'target', 'x64',
+                             'qemu-kernel.bin')
+  os.chmod(kernel_path, 0o644)
+  pkey_path = os.path.join(fuchsia_resources_dir, '.ssh', 'pkey')
+  os.chmod(pkey_path, 0o400)
+  sharefiles_path = os.path.join(fuchsia_resources_dir, 'qemu-for-fuchsia',
+                                 'share', 'qemu')
+  drive_path = os.path.join(fuchsia_resources_dir, 'target', 'x64', 'fvm.blk')
+  os.chmod(drive_path, 0o644)
+  fuchsia_zbi = os.path.join(fuchsia_resources_dir, 'target', 'x64',
+                             'fuchsia.zbi')
+  initrd_path = os.path.join(fuchsia_resources_dir, 'fuchsia-ssh.zbi')
+  os.chmod(initrd_path, 0o644)
+
+  # Perform some more initiailization steps.
+  extend_fvm(fuchsia_resources_dir, drive_path)
+  add_keys_to_zbi(fuchsia_resources_dir, initrd_path, fuchsia_zbi)
+
+  # TODO(flowerhack): Add a mechanism for choosing portnum dynamically.
+  portnum = '56339'
+
+  # yapf: disable
+  qemu_command = [
+      qemu_path,
+      '-m', '2048',
+      '-nographic',
+      '-kernel', kernel_path,
+      '-initrd', initrd_path,
+      '-smp', '4',
+      '-drive', 'file=' + drive_path + ',format=raw,if=none,id=blobstore',
+      '-device', 'virtio-blk-pci,drive=blobstore',
+      '-monitor', 'none',
+      '-append', '"kernel.serial=legacy TERM=dumb"',
+      '-machine', 'q35',
+      '-enable-kvm',
+      '-display', 'none',
+      '-cpu', 'host,migratable=no',
+      '-netdev',
+      ('user,id=net0,net=192.168.3.0/24,dhcpstart=192.168.3.9,'
+       'host=192.168.3.2,hostfwd=tcp::') + portnum + '-:22',
+      '-device', 'e1000,netdev=net0,mac=52:54:00:63:5e:7b',
+      '-L', sharefiles_path
+  ]
+  # yapf: enable
+
+  # Fuzzing jobs that SSH into the QEMU VM need access to these env vars.
+  environment.set_value('FUCHSIA_PKEY_PATH', pkey_path)
+  environment.set_value('FUCHSIA_PORTNUM', portnum)
+
+  # Finally, launch QEMU.
+  qemu_process = new_process.ProcessRunner(qemu_command[0], qemu_command[1:])
+  # TODO(flowerhack): Implement teardown logic.
+  qemu_process.run()
+
+
+def initialize_resources_dir():
+  """ Download Fuchsia QEMU resources from GCS bucket. """
+  resources_dir = environment.get_value('RESOURCES_DIR')
+  if not resources_dir:
+    raise errors.FuchsiaConfigError('Could not find RESOURCES_DIR')
+  fuchsia_resources_dir = os.path.join(resources_dir, 'fuchsia')
+
+  shell.create_directory(fuchsia_resources_dir)
```

Comment:
should this recreate every time it runs? Since we're redownloading things every time anyway, may be better to start from a clean slate?

Reply:
set recreate=True